datasets/gb/fcdo_sanctions/crawler.py

```diff
@@ -0,0 +1,188 @@
+        h.apply_name(entity, name_tag.text)
+    # Add phone numbers
+    for phone_number in designation.iterfind(".//PhoneNumbers//PhoneNumber"):
+        entity.add("phone", phone_number.text)
+    # Add email addresses
+    for email in designation.iterfind(".//EmailAddresses//EmailAddress"):
+        entity.add("email", email.text)
+    # Add addresses
+    for address in designation.iterfind(".//Addresses//Address"):
+        address_text = ""
+        for line in address.iter(tag=etree.Element):
+            address_text += line.text.strip() + "\n"
+        address_country = address.findtext(".//AddressCountry")
+        entity.add(
+            "address",
+            h.make_address(context, address_text.strip(), country=address_country),
+        )
+
+
+def make_person(context: Context, designation: etree.Element, entity):
+    individual = designation.find(".//IndividualDetails//Individual")
+    if individual is None:
+        unique_id = designation.find("UniqueID")
+        context.log.info(
+            f"No person details found for sanction with ID: {unique_id.text if unique_id is not None else 'Unknown ID'}"
+        )
+        return
+    # Add the date of birth
+    for dob in individual.iterfind(".//DOBs//DOB"):
+        entity.add("birthDate", h.parse_date(dob.text, formats=["%d/%m/%Y"]))
+    # Add titles
+    entity.add("title", individual.findtext(".//Title"))
+    # Add birthplace
+    for details in individual.iterfind(".//BirthDetails//Location"):
+        location = ""
+        if town := details.findtext(".//TownOfBirth"):
+            location += town + "\n"
+            entity.add("birthPlace", town)
+        if country := details.findtext(".//CountryOfBirth"):
+            location += country
+        entity.add("birthPlace", location)
+    # Add nationalities
+    for nationality in individual.iterfind(".//Nationalities//Nationality"):
+        entity.add("nationality", nationality.text)
+    # Add positions
+    for position in individual.iterfind(".//Positions//Position"):
+        entity.add("position", position.text)
+
+
+def make_ship(context: Context, designation: etree.Element, entity):
+    ship = designation.find(".//ShipDetails//Ship")
+    if ship is None:
+        unique_id = designation.find("UniqueID")
+        context.log.info(
+            f"No ship details found for sanction with ID: {unique_id.text if unique_id is not None else 'Unknown ID'}"
+        )
+        return
+    # Add the imonumber
+    imo_numbers = []
+    for imo in ship.iterfind(".//IMONumbers//IMONumber"):
+        imo_numbers.append(imo.text)
+    if len(imo_numbers) == 0:
+        raise Exception("No IMO number found in ship designation")
+    entity.id = imo_numbers[0]
+    entity.add("imoNumber", imo.text)
+    # Add the owners
+    for owner in ship.iterfind(".//CurrentOwnerOperators//CurrentOwnerOperator"):
+        owner_entity = context.make("Organization")
+        owner_entity.id = context.make_slug(owner.text)
```

Comment:
What purpose does the `named` serve here?

Reply:
Basically just to establish a sub-namespace, so that we know which keys are uniqueIDs and which ones are just name-based stubs. It's not important :)